lib/Dialect/TritonGPU/Transforms/Pipeline.cpp

```diff
@@ -1611,10 +1620,10 @@ void PipelinePass::asyncLaunchDots(scf::ForOp forOp) {
             valid = false;
 
           // C should be a block argument
-          auto CArg = dotOp.getOperand(2).dyn_cast<BlockArgument>();
-          if (!CArg || !CArg.hasOneUse())
+          auto CArg = dotOp.getOperand(2);
+          auto depend = dependOnTensorBlockArgument(dotOp.getOperand(2));
```

Comment:
Maybe we need check more conditions here.
For example
```
for (%0, %1, %2) {
    %3 = add %1, %2
    %4 = dot %0, %1, %3
    .... some other ops
    scf.yield %x, %y, %4
}
```
It is a undefined behavior if we async launch the DotOp in the loop. Because at the time point we issue the AddOp, its operand %2  will be not ready yet.

Reply:
Actually it only accept conditions

1. DotOp does not depends on operations with async semantics.
2. DotOp depends on operations with async semantics, and with proper sync/fence operations between them.
3. DotOp depends on itself, as a corner case of 2.